### PR TITLE
Clarify return value of `get_dependencies()`

### LINKED ISSUE #79300: `ResourceLoader::get_dependencies` breaking change in 4.1
### Godot version

v4.1.stable.official [970459615]

### System information

Godot v4.1.stable - Windows 10.0.19045 - Vulkan (Forward+) - dedicated Radeon (TM) RX 470 Graphics (Advanced Micro Devices, Inc.; 31.0.12027.9001) - AMD FX(tm)-8300 Eight-Core Processor (8 Threads)

### Issue description

`ResourceLoader::get_dependencies` returns a `PackedStringArray` with the resource's dependencies.

In **4.0.3**, each string is in the following format:
**uid://dvddbduy2au8w**.

But in **4.1**, each string is in the following format:
**uid://dvddbduy2au8w::::res://scenes/main_menu/main_menu_btn_pressed_style.tres**.

The change causes some errors that did not happen previously:

1. When fetching dependencies recursivelly, using the new output as input to `ResourceLoader::get_dependencies`;
2. When trying to load the new ouput with ``ResourceLoader::load``.

### Steps to reproduce

Run the **main** scene of the MRP.

### Minimal reproduction project

[get_dependencies.zip](https://github.com/godotengine/godot/files/12004931/get_dependencies.zip)

### PR DESCRIPTION
**Pull request by KoBeWi**
Closes #79300

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Comment by akien-mga** [comment]
Could you also add a mention to this in the "Changed" section of the 4.0 changelog in `CHANGELOG.md`? Linking the PR which changed this method.

**Comment by KoBeWi** [comment]
How it should be described? The PR title only mentions the fixed issue.

File: CHANGELOG.md [hunk 1]
```diff
@@ -167,6 +167,7 @@ See the [release announcement](https://godotengine.org/article/godot-4-1-is-here
 - Replace invalid characters in node names by underscores instead of removing them ([GH-75760](https://github.com/godotengine/godot/pull/75760)).
 - Optimize `Object::get_class_name` ([GH-75797](https://github.com/godotengine/godot/pull/75797)).
 - Ensure `RID`, `Callable`, and `Signal` are stored as strings ([GH-78517](https://github.com/godotengine/godot/pull/78517)).
+- The strings returned by `ResourceLoader::get_dependencies()` now include paths in addition to UIDs ([GH-73131](https://github.com/godotengine/godot/pull/73131)).
```
Comment (akien-mga) [inline CHANGELOG.md:170]: The description looks good. This should be moved as first in the list, we order them by PR number.

**Review by akien-mga — APPROVED** [review]

**Comment by YuriSizov** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.1.1.

**Pull request merged**